### ISSUE TITLE
config names that contain -<number> format are not recognized

### ISSUE THREAD

**Issue by amylashley** [sksamuel/hoplite#328]
I have a config variable of the format `variable-2` and when I run the app I receive an error: 
```
Caused by: com.sksamuel.hoplite.ConfigException: Error loading config because:

...

            - 'variable2': Missing from config
	at com.sksamuel.hoplite.ConfigLoader$returnOrThrow$1.invoke(ConfigLoader.kt:190)
	at com.sksamuel.hoplite.ConfigLoader$returnOrThrow$1.invoke(ConfigLoader.kt:187)
	at com.sksamuel.hoplite.fp.ValidatedKt.getOrElse(Validated.kt:98)
	at com.sksamuel.hoplite.ConfigLoader.returnOrThrow(ConfigLoader.kt:187)
```

I have to reformat it to `variable2` for it to work.

**Comment by sksamuel** [comment]
If you have `variable-2` in yml, what's the name of the field in the data class. Is it `variable2` ?

**Comment by amylashley** [comment]
Yes, that would be the variable name. A more specific example of my actual variable format (I tried to obfuscate my code) would be the config key `variable-name-2` with a corresponding variable called `variableName2`.

**Comment by sksamuel** [comment]
Fixed in 2.5.1
